fix(format): honor jsxEverywhere setting in variant-specific formatter rules (#7288)
biomejs/biome: .changeset/keep-jsxeverywhere-variant.md

---
"@biomejs/biome": patch
---

> Fixed [#7286](https://github.com/biomejs/biome/issues/7286): Files are now formatted with JSX behavior when jsxEverywhere is set

Reviewer: ```suggestion
Fixed [#7286](https://github.com/biomejs/biome/issues/7286). Files are now formatted with JSX behavior when `javascript.parser.jsxEverywhere` is explicitly set.
```

> Previously, this flag was only used for parsing, but not for formatting, which resulted in weird formatting of conditional expressions
> when a JSX file used the `.js` file extension.

Reviewer: ```suggestion
Previously, this flag was only used for parsing, but not for formatting, which resulted in incorrect formatting of conditional expressions when JSX syntax is used in `.js` files.
```
